code/modules/client/preference/loadout/loadout_tgui.dm

```diff
@@ -0,0 +1,61 @@
+GLOBAL_LIST_EMPTY(gear_tgui_info)
+
+/datum/ui_module/loadout
+	name = "Loadout"
+
+/datum/ui_module/loadout/ui_state(mob/user)
+	return GLOB.always_state
+
+/datum/ui_module/loadout/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "Loadout", name)
+		ui.set_autoupdate(FALSE)
+		ui.open()
+
+/datum/ui_module/loadout/ui_data(mob/user)
+	var/list/data = list()
+	data["gear_slots"] = user.client.prefs.build_loadout()
```

Comment:
Any possibility client could be null through these procs?

Reply:
I think not, but will add come checks